### PR TITLE
[Warnings] Make [p]warnings usable on base of permissions

### PR DESCRIPTION
**Pull request by Dav-Git**
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
Makes [p]warnings work with permissions and introduces [p]mywarnings with the functionality described in #2900

### CODE REVIEW

**Comment by Jackenmen** [comment]
Changelog contains wrong command names + I think changelogs should be separated into 2 files - 2900.enhance.1.rst and 2900.enhance.2.rst (though I heard today it might not be necessary so someone please confirm this)

**Comment by Dav-Git** [comment]
Wasn't sure on 2 files. I thought 1 file works

File: redbot/cogs/warnings/warnings.py [hunk 1]
```diff
@@ -342,36 +341,48 @@ def __init__(self, bot: Red):
...
-        appropriate permissions.
-        """
+        """List the warnings for the specified user."""
         if user is None:
-            user = ctx.author
-        else:
-            if not await is_admin_or_superior(self.bot, ctx.author):
-                return await ctx.send(
-                    warning(_("You are not allowed to check warnings for other users!"))
+            await ctx.send(_("Please provide a user you want to check the warnings for."))
```
Comment (mikeshardmind) [inline redbot/cogs/warnings/warnings.py]: This should either return here, or be removed entirely and remove the optional nature of the user argument (probably the latter, but either would be fine)

File: redbot/cogs/warnings/warnings.py [hunk 2]
```diff
@@ -342,36 +341,48 @@ def __init__(self, bot: Red):
...
+
+        msg = ""
+        member_settings = self.config.member(user)
```
Comment (mikeshardmind) [inline redbot/cogs/warnings/warnings.py:356]: This doesn't properly account for the possibility a user id was passed without it being to a current user (and needs to)

**Review by mikeshardmind — CHANGES_REQUESTED** [review]

File: redbot/cogs/warnings/warnings.py (hunk 1 above)
Comment (Dav-Git) [reply redbot/cogs/warnings/warnings.py]: Would removing the "Optional" part of the argument fix this?

File: redbot/cogs/warnings/warnings.py (hunk 2 above)
Comment (Dav-Git) [reply redbot/cogs/warnings/warnings.py:356]: Honestly I have no idea how to go about this.

File: redbot/cogs/warnings/warnings.py (hunk 1 above)
Comment (jack1142) [reply redbot/cogs/warnings/warnings.py]: Yes, making the argument required and removing handling for `user is None` would fix this.

File: redbot/cogs/warnings/warnings.py (hunk 2 above)
Comment (Jackenmen) [reply redbot/cogs/warnings/warnings.py:356]: The removed handling here was doing that before:
https://github.com/Cog-Creators/Red-DiscordBot/pull/3327/files#diff-bd66023271557007376afcf648f3b3ffL363-L368

File: redbot/cogs/warnings/warnings.py (hunk 1 above)
Comment (Jackenmen) [reply redbot/cogs/warnings/warnings.py]: Yes, making the argument required and removing handling for `user is None` would fix this.
(adding again, cause I accidentally removed the comment)

**Comment by Dav-Git** [comment]
I think that's what you had in mind?

File: redbot/cogs/warnings/warnings.py [hunk 3]
```diff
@@ -342,36 +341,52 @@ def __init__(self, bot: Red):
...
+    async def mywarnings(self, ctx: commands.Context):
+        """List warnings for yourself."""
+
+        user = ctx.author
+
+        msg = ""
+        member_settings = self.config.member(user)
+        async with member_settings.warnings() as user_warnings:
+
             try:
```
Comment (mikeshardmind) [inline redbot/cogs/warnings/warnings.py]: Not exactly what I meant... 

The git diff here is doing some slight disfavors in discussing review as it's recognizing the code for `mywarnings` now matches the old `warnings` more closely.

The handling for if a user id not a user was passed isn't needed in `mywarnings` but is needed in `warnings` (prior to interacting with config)

**Review by mikeshardmind — APPROVED** [review]
Thanks!

**Pull request merged**